server/src/main/java/tabletop/controllers/TournamentController.java

```diff
@@ -0,0 +1,125 @@
+package tabletop.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.bind.annotation.*;
+import tabletop.domain.game.wininformation.BasicWinInformation;
+import tabletop.domain.game.wininformation.SingleWinInformation;
+import tabletop.domain.match.tournament.Pair;
+import tabletop.domain.match.tournament.Tournament;
+import tabletop.domain.match.tournament.TournamentFinalResult;
+import tabletop.domain.match.tournament.TournamentType;
+import tabletop.domain.user.User;
+import tabletop.services.TournamentService;
+
+import java.util.*;
+
+@RestController
+@RequestMapping(value = "/tournament")
+public class TournamentController {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TournamentController.class);
+
+    private TournamentService tournamentService;
+
+    public TournamentController(TournamentService tournamentService) {
+        this.tournamentService = tournamentService;
+    }
+
+    @RequestMapping(value = "/types", method = RequestMethod.GET)
+    public List<TournamentType> getTournamentTypesList() {
+        List<TournamentType> tournamentTypes = new ArrayList<>();
+        tournamentTypes.addAll(Arrays.asList(TournamentType.values()));
+        return tournamentTypes;
+    }
+
+    @RequestMapping(value = "/finished", method = RequestMethod.GET)
+    public Collection<Tournament> getFinishedTournaments() {
+        return tournamentService.getFinishedTournaments();
+    }
+
+    @RequestMapping(value = "/show/{tournamentid}", method = RequestMethod.GET)
+    public Tournament getTournamentById(@PathVariable("tournamentid") Long tournamentid) {
+        LOGGER.debug("Return tournament with id: {}", tournamentid);
+        return tournamentService.getTournamentById(tournamentid);
+    }
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST, consumes = "application/json")
+    public void createTournament(@RequestBody Tournament tournament) {
```

Comment:
is this method for tests only? you can only create a tournament while creating an event